src/main/kotlin/org/openrewrite/kotlin/KotlinTypeSignatureBuilder.kt

```diff
@@ -194,7 +220,7 @@ class KotlinTypeSignatureBuilder(private val firSession: FirSession, private val
             is FirResolvedTypeRef -> classSignature(type.type)
             is FirResolvedQualifier -> convertClassIdToFqn(type.classId)
             else -> {
-                ""
+                throw UnsupportedOperationException("Unsupported class type: ${type.javaClass.name}")
```

Comment:
Each of these blank signatures is a mistake and can cause type issues since the first cached type will be returned for each type that returns a blank signature.